rfcs/0166-nix-formatting.md

```diff
@@ -0,0 +1,1182 @@
+
+*Bad code does not deserve good formatting.*
+
+## Detailed design
+
+### Standard Nix format
+
+The _standard Nix format_ defines the officially recommended way that Nix code should be formatted.
+
+The initial version of the standard Nix format is defined in a section towards the end:
+
+[Initial standard Nix format](#initial-standard-nix-format).
+
+Significant changes to the standard Nix format must go through another RFC.
+
+The latest version of the standard Nix format must be in a file on the main branch of the [official Nix formatter](#official-nix-formatter).
+
+### Nix format team
+
+The _Nix format team_ is established, it has the responsibility to
+- Maintain the [official Nix formatter](#official-nix-formatter)
+- Regularly [reformat Nixpkgs](#reformat-nixpkgs) with it
+
+See the linked sections for more information.
+
+Initially the team consists of these members:
+- @piegames (author of this RFC, shepherd of the original formatting RFC)
+- @infinisil (from Tweag, co-author of this RFC, shepherd of the original formatting RFC)
+- @tomberek (from Flox, shepherd of the original formatting RFC)
+- @0x4A6F (shepherd of the original formatting RFC)
+- @Sereja313 (from Serokell, original sponsors of nixfmt)
+
+Team member updates are left for the team itself to decide.
+
+### Official Nix formatter
+
+The Nix format team is given the authority and responsibility of
+creating and maintaining the _official Nix formatter_ implementation.
+This is a repository in the NixOS GitHub organisation.
+The repository will initially be based on [this nixfmt pull request](https://github.com/serokell/nixfmt/pull/118).
+This pull request has been developed along with this RFC and is already reasonably close to the proposed initial standard format.
+
+Any release of the official Nix formatter must conform to the latest version of the [standard Nix format](#standard-nix-format).
+
+The latest release of the official Nix formatter should support the Nix language syntax of the latest Nix release.
+The Nix format team should be consulted before the Nix language syntax is changed.
+
+### Reformat Nixpkgs
+
+For formatting Nixpkgs itself, a pinned release version of the official Nix formatter must be used.
+CI must generally enforce all files in Nixpkgs to be formatted with this version at all times.
+Automatically generated files may be handled differently, see [the next section](#automatically-generated-files).
+
+For every bump of the pinned formatter,
+the files of Nixpkgs must thus be re-formatted accordingly.
+Commits that reformat Nixpkgs will be added to `.git-blame-ignore-revs`,
+which can then be [ignored in tooling](#git-blames).
+The Nix format team is responsible for this task.
+
+In order to minimize conflicts especially when back-porting,
+the pinned formatter should preferably only be updated shortly before the release branch-off.
+This should be done in coordination with the NixOS release managers.
```

Comment:
This should be added to the release wiki to make sure it's never forgotten ;)

Reply:
Added in https://github.com/nix-rfc-101/rfcs/commit/e94e1e3d4c3368c4c4aead1c9fde7a6e1c53197f :)